staging/src/k8s.io/code-generator/cmd/validation-gen/validation.go

```diff
@@ -634,15 +722,39 @@ func (g *genValidations) emitValidationForType(c *generator.Context, inType *typ
 		if t.Kind == types.Struct || t.Kind == types.Alias {
 			// If this field is another type, call its validation function.
 			// Checking for nil is handled inside this call.
-			g.emitCallToOtherTypeFunc(c, t, "val", valPath, valIsPtr, sw)
+			g.emitCallToOtherTypeFunc(c, t, valIsPtr, valSW)
 		} else {
 			// No need to go further.  Struct- or alias-typed fields might have
 			// validations attached to the type, but anything else (e.g.
 			// string) can't, and we already emitted code for the field
 			// validations.
 		}
 
-		sw.Do("}\n", nil)
+		kName, vName := "_", "_"
+		if keyBuf.Len() > 0 {
+			kName = "key"
+		}
+		if valBuf.Len() > 0 {
+			vName = "val"
+		}
+		if keyBuf.Len()+valBuf.Len() > 0 {
```

Comment:
nit: Move `kName, vName` initialization into the `if keyBuf.Len()+valBuf.Len() > 0` clause?

Reply:
I have to touch this in a subsequent commit, so can we defer changes here?